tests/test_scheduler.py

```diff
@@ -0,0 +1,105 @@
+import os
+from unittest.mock import Mock, patch  # noqa: F401
+
+import pytest
+from mock import MagicMock
+
+from noisepy.seis.constants import AWS_BATCH_JOB_ID
+from noisepy.seis.scheduler import AWSBatchArrayScheduler, SingleNodeScheduler
+
+
+# Single Node Scheduler
+def test_single_node_scheduler_get_indices():
+    single_node_scheduler = SingleNodeScheduler()
+    items = ["a", "b", "c", "d"]
+    expected_indices = [0, 1, 2, 3]
+    indices = single_node_scheduler.get_indices(items)
+
+    assert indices == expected_indices
+
+
+def test_synchronize():
+    scheduler = SingleNodeScheduler()
+    result = scheduler.synchronize()
+
+    assert result is None
+
+
+# AWS Batch Array Scheduler
+
+
+# get_array_size test
+@patch("boto3.client")
+def test_get_array_size(mock_boto3_client):
+    boto3_response = {"jobs": [{"arrayProperties": {"size": 10}}]}
+
+    # Test 1 - valid job id format
+    # Mock the AWS_BATCH_JOB_ID environment variable with a valid format
+    os.environ["AWS_BATCH_JOB_ID"] = "job_id:123"
```

Comment:
nit: You can assign `job_id:123` to a constant (e.g. `VALID_JOB_ID`) so that you don't have to hard code it 4 times